### PR TITLE
[DS-354]: [Pyxis React] Split FormGrid in Grid and FormGrid

### PR DESCRIPTION
**Pull request by lauratrapani**
https://prima-assicurazioni-spa.myjetbrains.com/youtrack/issue/DS-354

## 💥 Breaking Changes
The `FormGrid` component doesn't export anymore the `Row` and `Col` components.

## ✨ Enhancements
A new `Grid` component which export also `Row` and `Col` has been created to permit the build of general-purpose page layouts based on CSS Grid.

### CODE REVIEW

File: packages/pyxis-react/src/components/FormGrid/FormGrid.tsx [hunk 1]
```diff
@@ -1,61 +1,43 @@
...
 );
 
 const Header: FC<ChildProps> = ({ children }) => <>{children}</>;
 const Content: FC<ChildProps> = ({ children }) => (
-  <FormGrid.Row>
-    <FormGrid.Col>
-      <div className={getClasses('medium')}>
+  <Grid.Row>
+    <Grid.Col>
+      <Grid>
```
Comment (emanuelemacri) [inline packages/pyxis-react/src/components/FormGrid/FormGrid.tsx:17]: The rowgap for fields is missing
```suggestion
      <Grid rowGap='medium'>
```

**Review by emanuelemacri — CHANGES_REQUESTED** [review]

File: packages/pyxis-react/src/components/FormGrid/FormGrid.test.tsx [hunk 2]
```diff
@@ -1,25 +1,34 @@
...
     const formGrid = screen.getByTestId(id);
     expect(formGrid).toBeInTheDocument();
     expect(formGrid).toHaveClass('grid');
   });
 
-  test('with large gap', () => {
-    render(<FormGrid id={id} rowGap="large"><>test</></FormGrid>);
-    const formGrid = screen.getByTestId(id);
-    expect(formGrid).toHaveClass('grid--gap-large');
+  test('can have a `Header` with the correct structure', () => {
```
Comment (alessiosantangelo) [inline packages/pyxis-react/src/components/FormGrid/FormGrid.test.tsx:14]:
```suggestion
  test('should contain an `Header` with the correct markup', () => {
```

File: packages/pyxis-react/src/components/FormGrid/FormGrid.test.tsx [hunk 3]
```diff
@@ -1,25 +1,34 @@
...
+    render(<FormGrid id={id}><FormGrid.Header><>test</></FormGrid.Header></FormGrid>);
+    const grid = screen.getByTestId(id);
+    expect(grid).toContainHTML('<div class="grid grid--gap-large" data-testid="test-grid" id="test-grid">test</div>');
   });
 
-  test('with custom classes', () => {
-    render(<FormGrid className="custom-class" id={id}><>test</></FormGrid>);
-    const formGrid = screen.getByTestId(id);
-    expect(formGrid).toHaveClass('custom-class');
+  test('can have a `Content` with the correct structure', () => {
```
Comment (alessiosantangelo) [inline packages/pyxis-react/src/components/FormGrid/FormGrid.test.tsx:20]:
```suggestion
  test('should have a `Content` with the correct markup', () => {
```

File: packages/pyxis-react/src/components/FormGrid/FormGrid.test.tsx [hunk 4]
```diff
@@ -1,25 +1,34 @@
...
+    render(<FormGrid id={id}><FormGrid.Content><>test</></FormGrid.Content></FormGrid>);
+    const grid = screen.getByTestId(id);
+    const row = grid.querySelector('.grid__row');
+    const column = grid.querySelector('.grid__row__column');
+    expect(row).toBeInTheDocument();
+    expect(column).toBeInTheDocument();
+  });
+
+  test('can have a `Footer` with the correct structure', () => {
```
Comment (alessiosantangelo) [inline packages/pyxis-react/src/components/FormGrid/FormGrid.test.tsx:29]:
```suggestion
  test('should have a `Footer` with the correct markup', () => {
```

File: packages/pyxis-react/src/components/Grid/Col/Col.test.tsx [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  });
+
+  test.each([1, 2, 3, 4, 5])(
+    'with `%s` span it should have proper classes',
+    (key) => {
+      render(<Grid.Col id={id} span={key as GridColSpan}>test</Grid.Col>);
+      const grid = screen.getByTestId(id);
+      expect(grid).toHaveClass(`grid__row__column--span-${key}`);
+    },
+  );
```
Comment (alessiosantangelo) [inline packages/pyxis-react/src/components/Grid/Col/Col.test.tsx:22]: Testing a cycle is often an anti-pattern.

File: packages/pyxis-react/src/components/Grid/Col/Col.test.tsx [hunk 6]
```diff
@@ -0,0 +1,29 @@
+import { render, screen } from '@testing-library/react';
+import Grid from 'components/Grid';
+import { GridColSpan } from './Col';
+
+const id = 'test-grid-col';
+
+describe('Grid.Col component', () => {
+  test('with default configuration it should mount with default classes', () => {
```
Comment (alessiosantangelo) [inline packages/pyxis-react/src/components/Grid/Col/Col.test.tsx:8]: Test should start with `should`

File: packages/pyxis-react/src/components/Grid/Grid.test.tsx [hunk 7]
```diff
@@ -0,0 +1,25 @@
+import { render, screen } from '@testing-library/react';
+import Grid from './Grid';
+
+const id = 'test-grid';
+
+describe('Grid component', () => {
+  test('with default configuration it should mount with default classes', () => {
```
Comment (alessiosantangelo) [inline packages/pyxis-react/src/components/Grid/Grid.test.tsx:7]:
```suggestion
  test('should have the correct markup when used with default configuration', () => {
```

File: packages/pyxis-react/src/components/Grid/Grid.test.tsx [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+    render(<Grid id={id}><>test</></Grid>);
+    const grid = screen.getByTestId(id);
+    expect(grid).toBeInTheDocument();
+    expect(grid).toHaveClass('grid');
+  });
+
+  test('with large gap', () => {
```
Comment (alessiosantangelo) [inline packages/pyxis-react/src/components/Grid/Grid.test.tsx:14]:
```suggestion
  test('should have a large Gap', () => {
```

**Review by alessiosantangelo — CHANGES_REQUESTED** [review]
Some tips: 
- always start tests with `should` instead of `with/when`. Test descriptions should define what response we expect
- do not test cycles, be imperative

File: packages/pyxis-react/src/components/Grid/Col/Col.test.tsx (hunk 5 above)
Comment (emanuelemacri) [reply packages/pyxis-react/src/components/Grid/Col/Col.test.tsx:22]: how? We do this in many cases where there are many variants to check

File: packages/pyxis-react/src/index.ts [hunk 9]
```diff
@@ -8,7 +8,7 @@ export { default as CheckboxGroup } from './components/CheckboxGroup';
 export { default as Counter } from './components/Counter';
 export { default as DateInput } from './components/DateInput';
 export { default as Form } from './components/Form';
-export { default as FormGrid } from './components/FormGrid';
+export { default as FormGrid } from './components/Grid';
```
Comment (Gsciarra) [inline packages/pyxis-react/src/index.ts:11]: Add `export { default as Grid } from './components/Grid';`

**Review by Gsciarra — APPROVED** [review]

File: packages/pyxis-react/src/components/Grid/Col/Col.test.tsx (hunk 5 above)
Comment (lauratrapani) [reply packages/pyxis-react/src/components/Grid/Col/Col.test.tsx]: [This card ](https://prima-assicurazioni-spa.myjetbrains.com/youtrack/issue/DS-378/Pyxis-React) has been created to solve the same problem in all components at once.

**Review by emanuelemacri — APPROVED** [review]
LGTM

**Review by alessiosantangelo — APPROVED** [review]

**Review by brianvanburken — APPROVED** [review]
LGTM :rocket:

**Pull request merged**